packages/prisma/src/prisma/schema.prisma

```diff
@@ -813,20 +945,147 @@ model Participation {
 
   sessionLeaderboards LeaderboardEntry[] @relation("SessionLeaderboards")
 
-  responses              QuestionResponse[]
-  detailResponses        QuestionResponseDetail[]
-  subscriptions          PushSubscription[]
-  completedMicroSessions String[]
-  bookmarkedQuestions    QuestionInstance[]
-  bookmarkedStacks       QuestionStack[]
+  responses               QuestionResponse[]
+  detailResponses         QuestionResponseDetail[]
+  subscriptions           PushSubscription[]
+  completedMicroSessions  String[]
+  bookmarkedQuestions     QuestionInstance[]
+  bookmarkedStacks        QuestionStack[]
+  bookmarkedElements      ElementInstance[]
+  bookmarkedElementStacks ElementStack[]
 
   createdAt DateTime @default(now())
   updatedAt DateTime @updatedAt
 
   @@unique([courseId, participantId])
 }
 
+enum LeaderboardType {
+  SESSION_BLOCK
+  SESSION
+  COURSE
+}
+
+model LeaderboardEntry {
+  id Int @id @default(autoincrement())
+
+  type LeaderboardType
+
+  score Int
+
+  participant   Participant @relation(fields: [participantId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+  participantId String      @db.Uuid
+
+  participation Participation?
+
+  sessionParticipation   Participation? @relation("SessionLeaderboards", fields: [sessionParticipationId], references: [id])
+  sessionParticipationId Int?
+
+  sessionBlock   SessionBlock? @relation(fields: [sessionBlockId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+  sessionBlockId Int?
```

Comment:
Do we link a leaderboard entry to a session block at any point? For the session leaderboard we use the link to the session and for the course leaderboard we use the link to the course. In the production DB, there are zero results with non NULL sessionBlockId

Reply:
I have no idea, I guess we do not persist anything during a block, it was conceptualized as a storage for LB snapshots after each block, but I guess that does not exist in any code :D we would need to go through code and check before we remove it here though